### PR TITLE
Add logs

### LINKED ISSUE #3: Table: Logs
Let's create a Transactions table using `chainwalkers.prod.harmony_txs` as the source data table.

**Two things to note:**
1. Data should be extracted from `tx:receipt:logs`. Logs has a decoded attribute that is dynamic per the contract log emitted. This should be a JSON variant column in the logs table we produce.
2. You will need to use [Snowflake's lateral flatten](https://docs.snowflake.com/en/sql-reference/functions/flatten.html) functionality to expand the tx.reciept.logs into individual rows.  

**To Do Next:**

- [x] Design the schema for this table
- [x] Get confirmation on schema
- [x] Build schema
- [x] Schema build review
- [x] Submit Pull Request

### PR DESCRIPTION
**Pull request by forgxyz**
Adjusted main dependency to be deduped_txs and included @antonyip's model for use of ref() function during testing. Model built successfully on initial and subsequent test.  
  
```
23:39:56  Found 3 models, 20 tests, 0 snapshots, 0 analyses, 182 macros, 0 operations, 0 seed files, 2 sources, 0 exposures, 0 metrics
23:39:57  Concurrency: 4 threads (target='default')
23:39:59  Finished running 1 incremental model in 3.82s.
23:40:00  Completed successfully
23:40:00  Done. PASS=1 WARN=0 ERROR=0 SKIP=0 TOTAL=1
```

### CODE REVIEW

File: dbt_project.yml [hunk 1]
```diff
@@ -31,8 +31,15 @@ clean-targets:         # directories to be removed by `dbt clean`
...
+    # example:
+      # +materialized: view
+
+# TODO configure project
 models:
   harmony:
-    # # Config indicated by + and applies to all files under models/example/
-    # example:
-    #   +materialized: view
+    core:
```
Comment (antonyip) [inline dbt_project.yml:41]: Lets remove this.. I think it automatically crawls the folders inside "source-paths" for the ymls and the yml's has the same pattern. Was something not working that you needed to add this?

If we do need this.. materialized should be defaulted to incremental as that should be our default mode..

File: macros/dedupe_utils.sql [hunk 2]
```diff
@@ -1,15 +1,15 @@
 {% macro deduped_blocks(table_name) -%}
     (
-        select 
+        select
```
Comment (antonyip) [inline macros/dedupe_utils.sql:3]: Not sure if you did this on purpose, or you had some tool to help you format these.. Might be good to share with the team for auto-formatters. :D

File: models/core/logs.sql [hunk 3]
```diff
@@ -0,0 +1,63 @@
+{{
+    config(
+        materialized='incremental',
+        unique_key = 'log_id',
+        tags=['core', 'logs'],
+        cluster_by=['block_timestamp']
+    )
+}}
+
+with base_txs as (
```
Comment (antonyip) [inline models/core/logs.sql]:
```suggestion
with

base_txs as (
```

**Review by antonyip — COMMENTED** [review]
Thanks @forgxyz! Seems like a great PR! Just a small question about the dbt_project.yml changes.

File: models/core/logs.sql [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+
+    select
+
+        block_id,
+        block_timestamp,
+        tx_id as tx_hash,
+        tx:receipt:logs as full_logs
+
+    from base_txs
+    where {{ incremental_load_filter("block_timestamp") }}
```
Comment (antonyip) [inline models/core/logs.sql]: Can we move this where clause to the base_txs? I know snowflake is pretty smart and probably can optimize it, but I think its okay to have our incremental "where" in as early as possible, together with the base clause.. (dbt style guide also doesn't have such an example) @chuxinh what do you think?

File: dbt_project.yml (hunk 1 above)
Comment (forgxyz) [reply dbt_project.yml:41]: I thought it was necessary 😅. Will remove

File: models/core/logs.sql (hunk 4 above)
Comment (forgxyz) [reply models/core/logs.sql]: Yep, I'm not against that. Still keeps the CTE dependency structure in tact.

**Review by antonyip — APPROVED** [review]
LGTM! Will merge after sync

**Comment by drakedanner** [comment]
Feel free to merge if it looks good @antonyip !

File: models/core/logs.sql [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+),
+
+logs as (
+
+  select
+
+    block_id,
+    block_timestamp,
+    tx_hash,
+    value:logIndex as event_index,
```
Comment (chuxinh) [inline models/core/logs.sql]: ideally you wanna parse the data type here too. more details here: https://docs.snowflake.com/en/user-guide/querying-semistructured.html
```suggestion
    value:logIndex::string as event_index,
```

Comment (forgxyz) [reply models/core/logs.sql]: 👍 saw the context on the Transfers PR. Thanks for the review, I'll add the type-casting to my table.  
  
@antonyip please wait to merge until the next commit.

**Comment by drakedanner** [comment]
Are we good to merge this one now @forgxyz @antonyip @chuxinh ?

**Pull request merged**